onnxruntime/core/optimizer/common_subexpression_elimination.cc

```diff
@@ -386,13 +436,14 @@ Status CommonSubexpressionElimination::ApplyImpl(Graph& graph, bool& modified, i
          output_index < end; ++output_index) {
       const NodeArg* output_def = node->OutputDefs()[output_index];
       auto equivalence_class = onnxruntime::make_unique<EquivalenceClass>(*node, input_values, output_index, discriminator);
+      
       auto* raw_ptr = equivalence_class.get();
 
       auto it = value_to_representative.find(raw_ptr);
       if (it == value_to_representative.end()) {
         unique_equivalence_classes.push_back(std::move(equivalence_class));
         it = value_to_representative.emplace_hint(it, raw_ptr,
-                                                  Representative{output_def, node_index, output_index});
+                                                  Representative{output_def, nullptr, node_index, output_index});
```

Comment:
The assert below ` ORT_ENFORCE(representative.output_index != kInvalidOutputIndex);` will fire for duplicate initializers. It should be converted to an if condition, and ideally the duplicated initializer should be removed from the graph.